Dockerfile

```diff
@@ -18,8 +18,8 @@ COPY main.go main.go
 COPY pkg/ pkg/
 
 ENV CGO_ENABLED=0
-ENV GOOS=$TARGETOS
-ENV GOARCH=$TARGETARCH
+ENV GOOS=${TARGETOS:-linux}
+ENV GOARCH=${TARGETARCH:-amd64}
 ENV GO111MODULE=on
 
 # Do an initial compilation before setting the version so that there is less to
```

Comment:
My testing worked with

`FROM --platform=${TARGETPLATFORM:-linux/amd64} gcr.io/distroless/static:nonroot`

Not sure if what you've done here with the GOOS and GOARCH will make it work without this.